nwb_conversion_tools/utils/neonwbephyswriter.py

```diff
@@ -0,0 +1,110 @@
+from abc import abstractmethod
+from distutils.version import StrictVersion
+
+import pynwb
+
+from .basenwbephyswriter import BaseNwbEphysWriter
+
+try:
+    import neo
+
+    if StrictVersion(neo.__version__) >= StrictVersion("0.10"):
+        HAVE_NEO = True
+    else:
+        HAVE_NEO = False
+except ImportError:
+    HAVE_NEO = False
+
+
+class NEONwbEphysWriter(BaseNwbEphysWriter):
+    """
+    Class to write a neo.RawIO or neo.IO versio>=0.10 to NWB
+
+    Parameters
+    ----------
+    object_to_write: neo.RawIO or neo.IO
+    stub: bool
+        whether to write a subset of recording extractor traces array as electrical series in nwbfile
+    stub_channels: list
+        channels to include when writing as stub
+    """
+
+    def __init__(self, object_to_write, stub=False, stub_channels=None):
+        assert HAVE_NEO
+        self.recording, self.sorting, self.event = None, None, None
+        BaseNwbEphysWriter.__init__(self, object_to_write, stub=stub, stub_channels=stub_channels)
+
+    @staticmethod
+    def supported_types():
+        assert HAVE_NEO
+        return (neo.rawio.baserawio.BaseRawIO, neo.io.baseio.BaseIO)
+
+    def add_to_nwb(self, nwbfile: pynwb.NWBFile, metadata=None, **kwargs):
+        # check what's in the neo object: analogsignals, spike trains, events and
+        # write recording, sorting, events accordingly
+        raise NotImplementedError
+
+    def add_recording(self, segment_index=0):
+        raise NotImplementedError
+
+    def add_sorting(self):
+        raise NotImplementedError
+
+    def add_waveforms(self):
+        raise NotImplementedError
+
+    def add_epochs(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def _make_recording_stub(self):
+        pass
+
+    @abstractmethod
+    def _make_sorting_stub(self):
+        pass
+
+    def _get_sampling_frequency(self):
+        pass
```

Comment:
should these be `NotImplementedError` instead of `pass`?

Reply:
I decided to have all the `get` methods return a `None` for now so `pass`. All the `add` methods which manipulate a variable, is a functionality that will be implemented in another PR.